packages/bot-skeleton/src/scratch/blocks/Text/text_statement.js

```diff
@@ -32,8 +33,43 @@ Blockly.Blocks.text_statement = {
             description: '',
         };
     },
+    onchange(event) {
+        if (!this.workspace || this.isInFlyout || this.workspace.isDragging()) {
+            return;
+        }
+
+        const stack_blocks = Blockly.getMainWorkspace().getBlockById(event.blockId);
+
+        if (event.type === Blockly.Events.END_DRAG) {
+            const surround_parent = this.getSurroundParent();
+            if (!surround_parent && !this.required_parent_id) {
+                return;
+            }
+            if (!this.required_parent_id && surround_parent?.type === this.required_parent_type) {
+                this.required_parent_id = surround_parent.id;
+            }
+            if (this.required_parent_id && (!surround_parent || surround_parent.id !== this.required_parent_id)) {
+                const all_blocks = this.workspace.getAllBlocks();
+                const original_parent = all_blocks.find(block => block.id === this.required_parent_id);
+
+                if (original_parent) {
+                    const first_block_in_stack = original_parent.getInputTargetBlock('STACK');
+
+                    if (first_block_in_stack) {
+                        const last_connection = original_parent.getLastConnectionInStatement('STACK');
+                        runIrreversibleEvents(() => {
+                            last_connection.connect(this.previousConnection);
+                        });
+                    } else {
+                        runIrreversibleEvents(() => {
+                            original_parent.getInput('STACK').connection.connect(stack_blocks.previousConnection);
+                        });
+                    }
+                }
+            }
+        }
```

Comment:
in your test link, if we get text_join from the toolbox and try to drag out text_statements outside of text_join we could not move them which is correct but if we refresh the page we could drag it outside of text_join 
could you please check it